### PR TITLE
Adding Rich Presence Support

### PR DESCRIPTION
**Pull request by ThatGuyJustin**

### CODE REVIEW

**Comment by b1naryth1ef** [comment]
You'll need to cleanup the flake8 issues;

```
disco/types/user.py:72:1: E302 expected 2 blank lines, found 1
disco/types/user.py:76:1: E302 expected 2 blank lines, found 1
disco/types/user.py:82:1: E302 expected 2 blank lines, found 1
disco/types/user.py:87:1: E302 expected 2 blank lines, found 1
disco/types/user.py:90:1: W293 blank line contains whitespace
disco/types/user.py:97:1: E302 expected 2 blank lines, found 1
disco/types/user.py:111:1: E302 expected 2 blank lines, found 1
```

As I mentioned to you OOB I'll take a look at this post-v1 (hopefully I can get an rc out the door this weekend).

**Comment by ThatGuyJustin** [comment]
Fixed!

**Comment by b1naryth1ef** [comment]
Can you rebase this PR off of the `staging/v1.0.0` branch please?

**Comment by ThatGuyJustin** [comment]
Done.

File: disco/types/user.py [hunk 1]
```diff
@@ -69,10 +71,48 @@ class Status(object):
...
+    join = Field(text)
+    spectate = Field(text)
+    match = Field(text)
+
+
+class Timestamps(SlottedModel):
+    start = Field(int)
+    end = Field(int)
+
+    def start_time(self):
```
Comment (b1naryth1ef) [inline disco/types/user.py]: Lets make these cached properties via `@cached_property`.

Comment (ThatGuyJustin) [reply disco/types/user.py]: Added

**Comment by b1naryth1ef** [comment]
Thanks!

**Pull request merged**